### ISSUE TITLE
Nested inline models always end up with 3 extra forms in the admin interface

### ISSUE THREAD

**Issue by darthmall** [wagtail/wagtail#5511]
### Issue Summary

If you create an inline model to an inline model, you'll always end up with 3 extra forms in the formset for everything after the first inline model. [This came up on StackOverflow](https://stackoverflow.com/questions/57510201/how-can-i-set-the-number-of-initial-empty-orderable-items-in-an-inlinepanel).

### Steps to Reproduce

Here's an example:

```python
class HomePage(Page):
    pass

    content_panels = [
        InlinePanel('features'),
    ]


class Feature(ClusterableModel, Orderable):
    page = ParentalKey(HomePage, related_name='features')
    heading = models.CharField()
    body = models.TextField()

    panels = [
        FieldPanel('heading'),
        FieldPanel('body'),
        InlinePanel('calls_to_action'),
    ]


class CallToAction(Orderable):
    feature = ParentalKey(Feature, related_name='calls_to_action')
    text = models.CharField()
    link = models.ForeignKey(
        Page,
        on_delete=models.CASCADE,
        related_name='+',
    )

    panels = [
        FieldPanel('text'),
        PageChooserPanel('link'),
    ]
```

With the above models, when you create a new `Feature` on the `HomePage`, it will automatically have 3 inline forms for `CallToAction`. 

The reason for this seems to be that `WagtailAdminModelFormMetaclass` doesn't override the `child_form` class method from `ClusterFormMetaclass`. So when a `WagtailAdminPageForm` or a `WagtailAdminModelForm` generates an inline form, that form is always a `ClusterForm`. And when that `ClusterForm` generates inline forms, it uses the `extra_form_count` from `ClusterFormMetaclass`, which is set to 3. So every inline model after the first level will end up with 3 extra forms.

I have a workaround for this, which involves creating a custom form class with its own metaclass.

```python
class HomePageFormMetaclass(WagtailAdminModelFormMetaclass):
    @classmethod
    def child_form(cls):
        return HomePageForm


class HomePageForm(WagtailAdminPageForm, metaclass=HomePageFormMetaclass):
    pass


class HomePage(page)
    base_form_class = HomePageForm

# etc.
```

But it seems to me that the ideal situation would be to make it easier to select how many extra inline models are generated for each `InlinePanel` the way Django does, rather than having it hard-coded as an attribute on a metaclass, which is very hard to change.

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: no

### Technical details

* Python version: 3.7.3.
* Django version: 2.2.4
* Wagtail version: 2.6.1

**Comment by gasman** [comment]
Thanks for the report @darthmall! Looks like we just need to add the snippet from https://github.com/wagtail/django-modelcluster/pull/107 to `WagtailAdminModelFormMetaclass`.

The standard within Wagtail is to set the extra-forms count to 0, since we have the 'add' button to dynamically add as many as desired. We can certainly consider making this overrideable, but that's best treated as a separate issue to the immediate bugfix (i.e. making nested formsets follow the expected Wagtail behaviour).

**Comment by lb-** [comment]
Resolved via 4001516 & #5566
